s3_to_redshift/asset_data_to_redshift.py

```diff
@@ -248,35 +268,69 @@ def report(data):
 
 report_stats['objects'] = len(objects_to_process)
 report_stats['incomplete_list'] = objects_to_process.copy()
+# list to keep track of objects
+good_objects = []
+path = ''
+
+
+# clean up the intermediate table
+bad_table_cleanup = r'''
+BEGIN;
+-- clean up the intermediate table with bad data
+{truncate_intermediate_table}
+COMMIT;
+'''.format(truncate_intermediate_table=truncate_intermediate_table)
 
 # process the objects that were found during the earlier directory pass
 for object_summary in objects_to_process:
     batchfile = destination + "/batch/" + object_summary.key
     goodfile = destination + "/good/" + object_summary.key
     badfile = destination + "/bad/" + object_summary.key
+    
+    spdb = RedShift.snowplow(batchfile) 
 
     # get the object from S3 and take its contents as body
     obj = client.get_object(Bucket=bucket, Key=object_summary.key)
 
-    # The file is an empty upload. Key to goodfile and continue
-    if(obj['ContentLength'] == 0):
-        logger.info('%s is empty, keying to goodfile and proceeding.',
+    # The file is an empty upload. Key to badfile and stop processing further.
+    if ((obj['ContentLength'] == 0) and (not empty_files_ok)):
+        logger.info('%s is empty, keying to badfile and no further processing.',
                      object_summary.key)
-        outfile = goodfile
+        outfile = badfile
+        #if there are any files in processed/good folder that were processed
```

Comment:
There are some linting issues in the PR. Add a space after the # 
https://www.python.org/dev/peps/pep-0008/#block-comments